core/src/com/unciv/logic/map/AncientRuins.kt

```diff
@@ -0,0 +1,147 @@
+                    civInfo.recordPickRuinBonus(RuinBonus.JOIN_SETTLER)
+                })
+            }
+            if (!blockedActions.contains(RuinBonus.JOIN_WORKER)) {
+                actions.add(RuinAction(RuinBonus.JOIN_WORKER) {
+                    performUnitJoin(civInfo, tile, Constants.worker)
+                    civInfo.recordPickRuinBonus(RuinBonus.JOIN_WORKER)
+                })
+            }
+        }
+
+        if(!unit.type.isCivilian() && !blockedActions.contains(RuinBonus.UPGRADE) && unit.getUnitToUpgradeTo(true) != unit.baseUnit()) {
+            actions.add(RuinAction(RuinBonus.UPGRADE) {
+                UnitActions.getUpgradeAction(unit, true)?.action?.invoke()
+                val unitName = unit.name;
+                civInfo.addNotification("An ancient tribe trains our [$unitName] in their ways of combat!", tile.position, Color.RED)
+                civInfo.recordPickRuinBonus(RuinBonus.UPGRADE)
+            })
+        }
+
+        if(!blockedActions.contains(RuinBonus.CULTURE)) {
+            actions.add(RuinAction(RuinBonus.CULTURE) {
+                civInfo.statsForNextTurn.add(Stat.Culture, 20f)
+                civInfo.addNotification("You have found cultural artifacts which awe your citizens! You have received 20 Culture!", tile.position, Color.PINK)
+                civInfo.recordPickRuinBonus(RuinBonus.CULTURE)
+            })
+        }
+
+        if (!blockedActions.contains(RuinBonus.GOLD)) {
+            actions.add(RuinAction(RuinBonus.GOLD) {
+                val amount = listOf(25, 60, 100).random(randomSeed)
+                civInfo.gold += amount
+                civInfo.addNotification("We have found a stash of [$amount] gold in the ruins!", tile.position, Color.GOLD)
+                civInfo.recordPickRuinBonus(RuinBonus.GOLD)
+            })
+        }
+
+        // Map of the surrounding area
+        if (!blockedActions.contains(RuinBonus.MAPS)) {
+            actions.add(RuinAction(RuinBonus.MAPS) {
+                val revealCenter = tile.getTilesAtDistance(ANCIENT_RUIN_MAP_REVEAL_OFFSET).toList().random(randomSeed)
+                val tilesToReveal = revealCenter
+                        .getTilesInDistance(ANCIENT_RUIN_MAP_REVEAL_RANGE)
+                        .filter { Random.nextFloat() < ANCIENT_RUIN_MAP_REVEAL_CHANCE }
+                        .map { it.position }
+                civInfo.exploredTiles.addAll(tilesToReveal)
+                civInfo.addNotification("We have found a crudely-drawn map in the ruins!", tile.position, Color.RED)
+                civInfo.recordPickRuinBonus(RuinBonus.MAPS)
+            })
+        }
+
+    }
+
+    private fun performUnitJoin(civInfo: CivilizationInfo, tile: TileInfo, chosenUnit: String) {
+        civInfo.placeUnitNearTile(tile.position, chosenUnit)
+        civInfo.addNotification("A [$chosenUnit] has joined us!", tile.position, Color.BROWN)
+    }
+
+    fun doRuinsBonus() {
+        if (unit.name == "Pathfinder") {
```

Comment:
Again, we should act according the the unit unique, not according to the unit name